code/lib/core-common/src/utils/load-main-config.ts

```diff
@@ -18,5 +20,51 @@ export async function loadMainConfig({
     delete require.cache[mainJsPath];
   }
 
-  return serverRequire(mainJsPath);
+  try {
+    const out = await serverRequire(mainJsPath);
+    return out;
+  } catch (e) {
+    if (!(e instanceof Error)) {
+      throw e;
+    }
+    if (e.message.match(/Cannot use import statement outside a module/)) {
+      const a = relative(process.cwd(), mainJsPath);
+      const line = e.stack?.match(new RegExp(`${a}:(\\d+):(\\d+)`))?.[1];
+      const message = [
+        `Storybook failed to load ${a}..`,
+        '',
+        `It looks like the file tried to load/import an ESM only module.`,
+        `Support for this is currently limited in ${a}.`,
+        `You can import ESM modules in your main file, but only as imperative imports.`,
```

Comment:
I worry that the meaning of `imperative imports` isn't immediately clear to users . Can you change the wording to be more accessible?

Also, can the error framework be used here?